errors.go

```diff
@@ -0,0 +1,43 @@
+package kable
+
+import (
+	"github.com/Kable-io/kable-go/internal/openapi"
+	"net/http"
+)
+
+// Error provides access to the body, status, and error on returned errors.
+type Error struct {
+	status int
+	body   []byte
+	error  string
+}
+
+// Error returns non-empty string if there was an error.
+func (e Error) Error() string {
```

Comment:
is this just a getter on `error.message`? let's maybe remove it?

From a consumer experience perspective, I'd love for `errors.go` to be almost unnoticeable -- small as possible until we've built out a real error-handling framework (retries, etc)